demos/python_demos/im2latex_demo/im2latex_demo.py

```diff
@@ -0,0 +1,252 @@
+    args.add_argument('--dec_st_h_t_layer', help='Optional. Decoder output key for dec_st_h_t. See README for details.',
+                      default='dec_st_h_t')
+    args.add_argument('--output_layer', help='Optional. Decoder output key for output. See README for details.',
+                      default='output')
+    args.add_argument('--output_prev_layer', help='Optional. Decoder input key for output_prev. See README for details.',
+                      default='output_prev')
+    args.add_argument('--logit_layer', help='Optional. Decoder output key for logit. See README for details.',
+                      default='logit')
+    args.add_argument('--tgt_layer', help='Optional. Decoder input key for tgt. See README for details.',
+                      default='tgt')
+    return parser
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s",
+                    level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+    log.info("Creating Inference Engine")
+    ie = IECore()
+    ie.set_config(
+        {"PERF_COUNT": "YES" if args.perf_counts else "NO"}, args.device)
+
+    # encoder part
+    encoder_model_xml = args.m_encoder
+    encoder_model_bin = os.path.splitext(encoder_model_xml)[0] + ".bin"
+    log.info("Loading encoder files:\n\t{}\n\t{}".format(
+        encoder_model_xml, encoder_model_bin))
+    encoder = ie.read_network(encoder_model_xml, encoder_model_bin)
+
+    # check if all layers are supported
+    supported_layers = ie.query_network(encoder, args.device)
+    not_supported_layers = [
+        l for l in encoder.layers.keys() if l not in supported_layers]
+    if len(not_supported_layers) != 0:
+        log.error("Following layers are not supported by the plugin for specified device {}:\n{}".
+                  format(args.device, ', '.join(not_supported_layers)))
+
+    # decoder part:
+    dec_step_model_xml = args.m_decoder
+    dec_step_model_bin = os.path.splitext(dec_step_model_xml)[0] + ".bin"
+
+    log.info("Loading decoder files:\n\t{}\n\t{}".format(
+        dec_step_model_xml, dec_step_model_bin))
+    dec_step = ie.read_network(dec_step_model_xml, dec_step_model_bin)
+
+    # check if all layers are supported
+    supported_layers = ie.query_network(dec_step, args.device)
+    not_supported_layers = [
+        l for l in dec_step.layers.keys() if l not in supported_layers]
+    if len(not_supported_layers) != 0:
+        log.error("Following layers are not supported by the plugin for specified device {}:\n{}".
+                  format(args.device, ', '.join(not_supported_layers)))
+    batch_dim, _, height, width = encoder.input_info['imgs'].input_data.shape
+    assert batch_dim == 1, "Demo only works with batch size 1"
+    target_shape = (height, width)
+    images_list = []
+    if os.path.isdir(args.input):
+        inputs = [os.path.join(args.input, inp)
+                  for inp in os.listdir(args.input)]
```

Comment:
```suggestion
        inputs = sorted(os.path.join(args.input, inp)
                  for inp in os.listdir(args.input))
```

(ensure consistent processing order)